shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -101,195 +101,58 @@ def _generate_state_file_string(self, gcp_backend_resource: GcpTfBackend):
-                    # GCP1G_MODEL:
-                    clp_details = self._validate_clp(api, gcp_backend_resource, logger)
-                    if clp_details.ResourceModelName == GCP1G_MODEL:
-                        print(clp_details.ResourceModelName)
-                        myactual=api.GetResourceDetails(clp_details.Name)
-                        clp_json_path = self._fill_backend_sercret_vars_data(myactual, GCP1G_MODEL)
-                        # account_keys = self._get_storage_keys(api, azure_backend_resource, clp_details)
-                        # model_prefix = ""
-                        # json_path = self._get_attrbiute_value_from_clp(
-                        #     clp_details.ResourceAttributes, model_prefix, CREDENTIALS_JSON_PATH)
-                        # self._handle_exception_logging(logger, f"json_pa {json_path}")
-                        # logger.exception(f"json_path: {json_path}") 
-                        # with open('C:\\out.txt', 'w') as f:
-                        #     with redirect_stdout(f):
-                        #         print('data')
-                        #         print(f"JSON path: {json_path}")
-                        #         print(f"gcp_backend_resource: {gcp_backend_resource.cloud_provider}")  
-                        # if json_path:
-                        # logger.exception(f"json_path: {clp_json_path}")
-                        # json = "C:\\Users\\CSadmin123456\\CS\\tf-gcp\\prod.json"
-                        # logger.exception(f"def: {json}")
-                        os.environ[GOOGLE_APPLICATION_CREDENTIALS] = clp_json_path
-                        # self._handle_exception_logging(logger, f"CLP: {clp_json_path}  DEF: {json}") 
-                        # os.environ[GOOGLE_APPLICATION_CREDENTIALS] = "C:\\Users\\CSadmin123456\\CS\\tf-gcp\\prod.json"
-            except Exception as e:
-                self._handle_exception_logging(logger, f"There was an issue accessing GCP. {e}")
-            bucket_name = gcp_backend_resource.bucket_name
-            if bucket_name:      
-                self._validate_bucket_exists(bucket_name, context)
-
-    def _get_attrbiute_value_from_clp(self, attributes, model_prefix, attribute_name) -> str:
-        for attribute in attributes:
-
-            if attribute.Name == f"{model_prefix}{attribute_name}":
-                return attribute.Value
-        return ""
-
+    def _create_gcp_session(self, context, project_id, logger):
+        # with LoggingSessionContext(context) as logger:
+        api = CloudShellSessionContext(context).get_api()
+        gcp_backend_resource = GcpTfBackend.create_from_context(context)
+        # try:
+        private_key = api.DecryptPassword(gcp_backend_resource.private_key).Value.replace("\\n", "\n")
+        email = api.DecryptPassword(gcp_backend_resource.client_email).Value
+        if not project_id:
+            self._raise_and_log(logger, "Project id must be filled")
```

Comment:
I recommend to move this 2 line to the beginning of the method. Because there is no point to initialize the api and decrypt the passwords if there is not project id. Makes sense?

Reply:
yes, agree.